src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrScanner.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+/// <summary>
+/// Class that scans the VMR for cloaked files that shouldn't be inside.
+/// </summary>
+public class VmrScanner : IVmrScanner
+{
+    private const string _vmrPreserveAttribute = "vmr-preserve";
```

Comment:
Please use `VmrPatchHandler.KeepAttribute` please

Reply:
Do we have some class where I can move that const to? That might be cleaner that making it public in the `VmrPatchHandler`

Reply:
Fair point. `VmrInfo` would be good for that

Reply:
I moved both `KeepAttribute` and `IgnoreAttribute` to it to be consistent